blog/migrations/0017_migrate_image_data_to_custom_image_model.py

```diff
@@ -0,0 +1,75 @@
+from django.conf import settings
+from django.db import migrations
+from django.db.models import ForeignKey
+from django.db.models.deletion import SET_NULL
+
+cover_photos = {}
+article_photos = {}
+
+
+def move_existing_image_data_to_custom_model(apps, _schema_editor):
+    wagtail_image_model = apps.get_model("wagtailimages", "image")
+    custom_image_model = apps.get_model("blog", "customimage")
+    for image in wagtail_image_model.objects.all().order_by("id"):
+        custom_image = custom_image_model(
+            title=image.title,
+            file=image.file,
+            width=image.width,
+            height=image.height,
+            created_at=image.created_at,
+            uploaded_by_user=image.uploaded_by_user,
+            tags=image.tags,
+            focal_point_x=image.focal_point_x,
+            focal_point_y=image.focal_point_y,
+            focal_point_width=image.focal_point_width,
+            focal_point_height=image.focal_point_height,
+            file_size=image.file_size,
+            file_hash=image.file_hash,
+        )
+        custom_image.save()
+
+
+def copy_foreign_key_relations_between_images_and_articles(apps, _schema_editor):
+    custom_image_model = apps.get_model("blog", "customimage")
+    blog_article_page_model = apps.get_model("blog", "blogarticlepage")
+    for blog_article_page in blog_article_page_model.objects.all():
+        cover_photos[blog_article_page.pk] = custom_image_model.objects.get(id=blog_article_page.cover_photo.id)
+        article_photos[blog_article_page.pk] = custom_image_model.objects.get(id=blog_article_page.article_photo.id)
+
+
+def assign_new_images_to_articles(apps, _schema_editor):
+    blog_article_page_model = apps.get_model("blog", "blogarticlepage")
+    for blog_article_page in blog_article_page_model.objects.all():
+        blog_article_page_model.cover_photo = cover_photos[blog_article_page.pk]
+        blog_article_page_model.article_photo = article_photos[blog_article_page.pk]
```

Comment:
I don't quite get it how it works - there is no save inside here, either changing `cover_photo` or `article_photo` on any instance

Reply:
You are absolutely right, not only does this function doesn't accomplish anything (notice that the fields are called from the **model** and not the **instance** as well!), but both functions above above appear to be absolutely unnecessary.
Here's what I think happens:
- The custom images are created in same order as original images, so they most likely share the PKs as well)
- The article model's field is altered to accept a different model, but the actual PK reference is still stored there
- Because those objects already exists, the field value is valid and points to a corresponding custom image model now.